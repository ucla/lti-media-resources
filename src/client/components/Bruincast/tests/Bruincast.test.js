/* eslint-disable no-throw-literal */

import React from 'react';
import axios from 'axios';
import { render, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
// Import axiosRetry from 'axios-retry';
import { Bruincast } from '../index';

jest.mock('axios');
// Jest.mock('axios-retry');
// const axiosRetryMod = { axiosRetry };

/**
 * Function that mock axios
 */
// async function axiosMock() {
//   throw {
//     response: {
//       status: 400,
//     },
//   };
// }

test('axios error handling in App', done => {
  // Const retrySpy = jest.spyOn(axiosRetryMod, 'axiosRetry');
  axios.get.mockImplementationOnce(() =>
    Promise.reject(
      new Error({
        response: {
          status: 400,
        },
      })
    )
  );
  const sampleCourse = { label: '20S-LOL-1', quarter: '20S', title: 'LOL' };
  const sampleWarning = '<p></p>';
  const sampleRetrieveWarning = jest.fn();
  const sampleUserid = 1;
  const sampleSetError = jest.fn(obj => obj);
  const bcastInstance = render(
    <Bruincast
      course={sampleCourse}
      warning={sampleWarning}
      retrieveWarning={sampleRetrieveWarning}
      userid={sampleUserid}
      setError={sampleSetError}
    />
  );
  // Expect(bcastInstance).toMatchSnapshot();
  expect(axios.get).toBeCalled();
  done();
});
