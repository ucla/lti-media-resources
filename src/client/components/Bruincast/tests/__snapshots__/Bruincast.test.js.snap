// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`axios error handling in App 1`] = `
<View
  borderColor="primary"
  display="auto"
  focusColor="info"
  focusPosition="offset"
  overflowX="visible"
  overflowY="visible"
  position="static"
  shouldAnimateFocus={true}
  withVisualDebug={false}
>
  <Heading
    border="none"
    color="inherit"
    level="h2"
  >
    Bruincast: LOL
  </Heading>
  <Alert
    isLiveRegionAtomic={false}
    liveRegionPoliteness="assertive"
    margin="x-small 0"
    open={true}
    renderCloseButtonLabel="Close"
    screenReaderOnly={false}
    timeout={0}
    transition="fade"
    variant="warning"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p></p>",
        }
      }
    />
  </Alert>
  <Text
    as="span"
    letterSpacing="normal"
    size="medium"
    wrap="normal"
  >
    Video recordings may take up to 24 hours before they are available.
     
    <Link
      color="link"
      href="#"
      iconPlacement="start"
      isWithinText={true}
    >
      Help
    </Link>
  </Text>
  <Tabs
    elementRef={[Function]}
    onRequestTabChange={[Function]}
    shouldFocusOnRender={false}
    tabOverflow="stack"
    variant="secondary"
  />
</View>
`;
